### PR TITLE
Feature/add tensorflowlite backend loader

### PR DESCRIPTION
**Pull request by maumonteroj**

### CODE REVIEW

File: r2i/tensorflowlite/loader.cc [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+std::shared_ptr<r2i::IModel> Loader::Load (const std::string &in_path,
+    r2i::RuntimeError &error) {
+
+  if (in_path.empty()) {
+    error.Set (RuntimeError::Code::WRONG_API_USAGE, "Received NULL path to file");
+    return nullptr;
+  }
+
+  std::ifstream graphdef_file;
+  graphdef_file.open (in_path, std::ios::binary | std::ios::ate);
```
Comment (michaelgruner) [inline r2i/tensorflowlite/loader.cc]: You are never closing the file.

File: tests/unit/r2i/tensorflowlite/loader.cc [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+  return fake_model;
+}
+
+TEST_GROUP (TensorflowliteLoader) {
+  r2i::RuntimeError error;
+
+};
+
+TEST (TensorflowliteLoader, WrongApiUsage) {
```
Comment (michaelgruner) [inline tests/unit/r2i/tensorflowlite/loader.cc]: Use a more explicit test name. Something like `EmptyString`

File: tests/unit/r2i/tensorflowlite/loader.cc [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  CHECK_TEXT (error.IsError(), error.GetDescription().c_str());
+  LONGS_EQUAL (r2i::RuntimeError::Code::FILE_ERROR, error.GetCode ());
+}
+
+TEST (TensorflowliteLoader, UnableToReadFile) {
+  std::shared_ptr<r2i::tensorflowlite::Loader> loader (new
+      r2i::tensorflowlite::Loader);
+
+  /* This will work as long as make check is run as a regular user */
+  std::string empty_string = "/root";
```
Comment (michaelgruner) [inline tests/unit/r2i/tensorflowlite/loader.cc]: No, this is highly non portable. Not every OS has the root concept.

File: tests/unit/r2i/tensorflowlite/loader.cc [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+TEST (TensorflowliteLoader, LoadInvalidFile) {
+  /* Setup */
+  incompatible_model = true;
+
+  std::shared_ptr<r2i::tensorflowlite::Loader> loader (new
+      r2i::tensorflowlite::Loader);
+
+  /* Attempt to load this file as a valid model */
+  auto model = loader->Load(__FILE__, error);
```
Comment (michaelgruner) [inline tests/unit/r2i/tensorflowlite/loader.cc]: This doesn't test an invalid file. The test is ran from another directory where this file doesn't exist. So this actually tests a file that doesn't exist.

File: tests/unit/r2i/tensorflowlite/loader.cc [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+  /* Teardown */
+  incompatible_model = false;
+}
+
+TEST (TensorflowliteLoader, LoadSuccess) {
+  /* Setup */
+  pass_fake_model = true;
+
+  std::shared_ptr<r2i::tensorflowlite::Loader> loader (new
```
Comment (michaelgruner) [inline tests/unit/r2i/tensorflowlite/loader.cc]: You create the loader with every test. Factor out to the setup method.

File: tests/unit/r2i/tflite/loader.cc [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+  /* Attempt to load this file as a valid model */
+  auto model = loader->Load(non_existent_file, error);
+
+  CHECK_TEXT (error.IsError(), error.GetDescription().c_str());
+  LONGS_EQUAL (r2i::RuntimeError::Code::FILE_ERROR, error.GetCode ());
+}
+
+TEST (TensorflowliteLoader, UnableToReadFile) {
+  /* This will work as long as make check is run as a regular user */
+  std::string empty_string = "/root";
```
Comment (michaelgruner) [inline tests/unit/r2i/tflite/loader.cc]: This is not portable. Create a `resources` dir within the test and create a file without permissions.

Comment (michaelgruner) [reply tests/unit/r2i/tflite/loader.cc]: Also, empty_string is not an appropriate name.

File: tests/unit/r2i/tflite/loader.cc [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    error.Set (r2i::RuntimeError::Code::INCOMPATIBLE_MODEL, "Incompatible model");
+  }
+
+  return error;
+}
+
+TEST_GROUP (TensorflowliteLoader) {
+  r2i::RuntimeError error;
+  std::shared_ptr<r2i::tflite::Loader> loader{new r2i::tflite::Loader};
+  std::string path;
```
Comment (michaelgruner) [inline tests/unit/r2i/tflite/loader.cc]: This is unused.

File: tests/unit/r2i/tflite/loader.cc [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+};
+
+TEST (TensorflowliteLoader, EmptyString) {
+  std::string empty_string = "";
+
+  /* Attempt to load this file as a valid model */
+  auto model = loader->Load(empty_string, error);
```
Comment (michaelgruner) [inline tests/unit/r2i/tflite/loader.cc]: Watch out! you are using the same error for all the tests. You **must** clean it for every test. Create a `setup()` function to do this.

File: tests/unit/r2i/tflite/loader.cc [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+  LONGS_EQUAL (r2i::RuntimeError::Code::FILE_ERROR, error.GetCode ());
+
+  /* Teardown */
+  incompatible_model = false;
+}
+
+TEST (TensorflowliteLoader, LoadSuccess) {
+  /* Setup */
+  pass_fake_model = true;
+  path = "sources/mobilenet_v1_1.0_224.tflite";
```
Comment (michaelgruner) [inline tests/unit/r2i/tflite/loader.cc]: I don't like the `sources` name. Maybe consider `resources` instead. Also, eventually I'd like a much smaller binary for this.

File: tests/unit/r2i/tflite/loader.cc [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+  auto model = loader->Load(non_existent_file, error);
+
+  CHECK_TEXT (error.IsError(), error.GetDescription().c_str());
+  LONGS_EQUAL (r2i::RuntimeError::Code::FILE_ERROR, error.GetCode ());
+}
+
+TEST (TensorflowliteLoader, UnableToReadFile) {
+  /* This will work as long as make check is run as a regular user */
+  std::string without_permissions_string =
+    "resources/file_without_permissions.tflite";
```
Comment (michaelgruner) [inline tests/unit/r2i/tflite/loader.cc]: This file was not uploaded.

File: r2i/tflite/loader.cc [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+
+std::shared_ptr<r2i::IModel> Loader::Load (const std::string &in_path,
+    r2i::RuntimeError &error) {
+
+  if (in_path.empty()) {
+    error.Set (RuntimeError::Code::WRONG_API_USAGE, "Received empty path to file");
+    return nullptr;
+  }
+
+  std::ifstream graphdef_file;
```
Comment (rrcarlosrodriguez) [inline r2i/tflite/loader.cc]: Move this definition at the beginning of the function

**Pull request merged**